### ISSUE TITLE
Does not install on PHP7

### ISSUE THREAD

**Issue by belgattitude** [bitExpert/prophiler-psr7-middleware#17]
Could not install with PHP7.1. 

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - fabfuel/prophiler 1.5.0 requires php ~5.4 -> your PHP version (7.1.5) does not satisfy that requirement.
    - fabfuel/prophiler 1.4.0 requires php ~5.4 -> your PHP version (7.1.5) does not satisfy that requirement.
    - bitexpert/prophiler-psr7-middleware v0.4.0 requires fabfuel/prophiler ^1.4|dev-feature/php7 -> satisfiable by fabfuel/prophiler[1.4.0, 1.5.0].
    - Installation request for bitexpert/prophiler-psr7-middleware ^0.4.0 -> satisfiable by bitexpert/prophiler-psr7-middleware[v0.4.0].
```

**Comment by belgattitude** [comment]
I could make it work with 

```shell
$ composer require --dev fabfuel/prophiler:dev-feature/php7 bitexpert/prophiler-psr7-middleware
```

I might have missed in the doc ? But can be usefull to write it.

**Comment by shochdoerfer** [comment]
Thanks for pointing out. In the composer.json we require ` "fabfuel/prophiler": "^1.4|dev-feature/php7"` which I thought would fix the problem. Maybe we should release a new version and specifically target  PHP 7.1. Will think about it.

**Comment by belgattitude** [comment]
Would be great ;)

Note that I couldn't make it work under zend-expressive 2.0 with 'programmatic pipelines'... At least from the provided doc, as a solution you can see: https://github.com/zendframework/zend-expressive-skeleton/issues/157

**Comment by shochdoerfer** [comment]
Sorry but I haven't really followed the expressive development. Can you elaborate a bit what changed and what needs to be changed in the middleware to make it work again? Mind providing a PR?

Ideally we could then release a version that is compatible with expressive 2.0 and another release which is only compatible with PHP 7.1.

**Comment by belgattitude** [comment]
@shochdoerfer,

expressive 2.0 have introduced the 'programmatic pipelines' (became default) in addition to the existing module based configuration. 

In other words, you probably need to update your [blog post](https://blog.bitexpert.de/blog/using-prophiler-with-zend-expressive/):

This part is valid for module based config:

> At first I had to register a pre_routing middleware in config/autoload/middleware-pipeline.local.php:
> ```php
> return [
>    'middleware_pipeline' => [[
>        'middleware' => bitExpert\Http\Middleware\Psr7\
>            Prophiler\ProphilerMiddleware::class,
>        'priority' => 11000,
>    ]]
>];
>```

But with programmatic pipelines, it must be defined in a file (just like for slim I guess)

> Adding them in the config/pipeline.php file:
> ```php
> $app->pipe(ErrorHandler::class);
> //... just after the ErrorHandler::class (must be the first one)
>
> $debug = $app->getContainer()->get('config')['debug'] ?? false;
> if ($debug) {
>    /* example with prophilermiddleware */
>    $prophiler  = new \Fabfuel\Prophiler\Profiler();
>    $toolbar    = new \Fabfuel\Prophiler\Toolbar($prophiler);
>    $middleware = new \bitExpert\Http\Middleware\Psr7\Prophiler\ProphilerMiddleware($toolbar);
>    $app->pipe($middleware);
> }
>
> $app->pipe(ServerUrlMiddleware::class);
> //... the rest of the pipeline
> ```


Actually there's no issue with your code... It's more about documentation to cover the two alternatives install on expressive.

**Comment by belgattitude** [comment]
@shochdoerfer,  look at the P/R, if it works with php5 it might be a solution for now (no breaking changes needed). If aliased the dep to 2.0:

```
"fabfuel/prophiler": "^1.4|dev-feature/php7 as 2.0",
```

So the dev-feature/php7 will have precedence over 1.4

**Comment by belgattitude** [comment]
I need more time to test, because for now, looks the request collector does not give anything under expressive 2.0.

**Comment by belgattitude** [comment]
@shochdoerfer should work now, please have a look at the P/R #18 for infos.

**Comment by shochdoerfer** [comment]
Does it make sense to include a default `config/pipeline.php` in this package? Is that the way it is supposed to work?

**Comment by belgattitude** [comment]
No, when people use programmatic pipelines, they're suppose to *code* the pipeline. See https://github.com/zendframework/zend-expressive-skeleton/blob/master/config/pipeline.php

So documenting is enough.

**Comment by shochdoerfer** [comment]
@belgattitude mind checking the updated docs please? Thanks. See [this](https://github.com/bitExpert/prophiler-psr7-middleware/pull/19) PR.

**Comment by shochdoerfer** [comment]
Where you able to fix the problem with the request collector?

**Comment by belgattitude** [comment]
Hey @shochdoerfer ,

haven't had the time yet... But I guess this week-end will be possible or Monday.

**Comment by shochdoerfer** [comment]
Fixed with #18
